### DISCUSSION TITLE
Individual counters for instructions under IntAlu

### DISCUSSION THREAD

**Discussion by RAZ4AMZ** [gem5/gem5#1802] [Q&A]
I was wondering how I would go about getting the individual counts for various instructions(primarily the instructions which are grouped together as IntAlu) as a part of a benchmarking effort.

I was able to figure out where the flags are defined for isInteger and so on, but I could not figure out where they are being set (expected it to be under decoder.h).

**Comment by powerjg** [comment]
For the minor and O3 CPUs that use [functional units](https://github.com/gem5/gem5/blob/stable/src/cpu/func_unit.cc) you could probably add stats to that code. This won't work for the simple CPUs since they don't have function units.

This seems like something that could be useful to others, so I would encourage you to upstream your code if you do add this statistic.

**Reply by RAZ4AMZ** [reply]
Yes, I was looking through the functional units in O3 CPUs which actually uses the isInteger flag internally. However, it seems that the flag is set based on the type of registers and not the actual op-code. So, I am currently confused as to how I would approach this problem.
